tajo-core/src/main/java/org/apache/tajo/ha/HdfsServiceTracker.java

```diff
@@ -458,21 +454,37 @@ public InetSocketAddress getMasterHttpInfo() throws ServiceTrackerException {
         throw new ServiceTrackerException("Active master base path must be a directory.");
       }
 
-      FileStatus[] files = fs.listStatus(activeMasterBaseDir);
       /* wait for active master from HDFS */
       int pause = conf.getIntVar(ConfVars.TAJO_MASTER_HA_CLIENT_RETRY_PAUSE_TIME);
       int maxRetry = conf.getIntVar(ConfVars.TAJO_MASTER_HA_CLIENT_RETRY_MAX_NUM);
       int retry = 0;
 
-      while (files.length < 2 && retry < maxRetry) {
+      FileStatus[] files = fs.listStatus(activeMasterBaseDir);
```

Comment:
It would be better if it is moved into the inner loop. If so, it will handle the case where the master entry is created during retrying.